File: 강의/김영한 강의들/복습/jpa/ex2-jpashop/src/main/java/dompoo/jpashop/domain/Member.java
package dompoo.jpashop.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Member {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Long id;
    private String name;
    private String city;
    private String street;
    private String zipcode;
}

File: 언어/kopring/src/main/kotlin/dompoo/kopringdemo/domain/Member.kt
package dompoo.kopringdemo.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.time.LocalDate
import java.time.Period

@Entity
class Member(
	@Column
	var username: String,
	@Column
	var birth: LocalDate
) {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	var id: Long = 0

	var age:Int = Period.between(birth, LocalDate.now()).years
}